var/spack/repos/builtin/packages/lz4/package.py

```diff
@@ -52,3 +52,20 @@ def install(self, spec, prefix):
         if self.run_tests:
             make('test')  # requires valgrind to be installed
         make('install', 'PREFIX={0}'.format(prefix))
+
+    # Without this method, lz4 installs with the wrong "dylib name" --
+    # this name should be $(spack location -i lz4)/lib/liblz4.1.dylib,
+    # but instead lists as "/usr/local/lib/liblz4.1.dylib". This
+    # information can be checked using "otool -L". To correct the
+    # dylib name, we have to run "install_name_tool -id".
+    @when('platform=darwin')
+    @run_after('install')
+    def fix_dylib_name(self):
```

Comment:
there is a Spack's function for that https://github.com/spack/spack/blob/develop/lib/spack/llnl/util/filesystem.py#L565-L600